src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TypeAnalysis.cs

```diff
@@ -25,6 +26,10 @@ internal sealed class TypeAnalysis
         internal readonly EventOpcode opcode = (EventOpcode)(-1);
         internal readonly EventTags tags;
 
+#if !ES_BUILD_STANDALONE
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
```

Comment:
This shouldn't be a suppress message. It should be `RequiresUnreferencedCode`

Reply:
Given the recursive call aspect of this issue, it's not possible to add the `RequiresUnreferencedCode`. Hence adding the `RequiresUnreferencedCode` to the Write Event methods

Reply:
I am not sure I follow. If you mark this method as `RequiresUnreferencedCode`, it will stop the linker from warning here, and instead it will start warning for all the callers of this code. Which is what we want.

If we add some new API that calls this method, it won't get a warning, because it is suppressed here. But if you change this to `RequiresUnreferencedCode`, the linker will ensure all calls to this code are either suppressed or annotated with `RequiresUnreferencedCode`.

Reply:
Incorporated all the other feedback and leaving this as a separate commit since the required `RequiresUnreferencedCode` annotations will be significant.

Reply:
> since the required RequiresUnreferencedCode annotations will be significant.

In my quick analysis, I see about 10-15 places that will need to be updated.

I don't believe addressing this comment should be a separate commit because this change in not correct by itself. Suppressing this warning means you could miss callsites that are calling into this method, and nullifying the linker's analysis.